server/src/main/java/org/elasticsearch/repositories/VerifyNodeRepositoryAction.java

```diff
@@ -8,149 +8,77 @@
 
 package org.elasticsearch.repositories;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
 import org.elasticsearch.cluster.node.DiscoveryNode;
-import org.elasticsearch.cluster.node.DiscoveryNodes;
 import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.threadpool.ThreadPool;
-import org.elasticsearch.transport.TransportChannel;
-import org.elasticsearch.transport.TransportException;
-import org.elasticsearch.transport.TransportRequest;
-import org.elasticsearch.transport.TransportRequestHandler;
-import org.elasticsearch.transport.TransportResponse;
-import org.elasticsearch.transport.TransportResponseHandler;
 import org.elasticsearch.transport.TransportService;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-import java.util.concurrent.CopyOnWriteArrayList;
-import java.util.concurrent.Executor;
-import java.util.concurrent.atomic.AtomicInteger;
 
-public class VerifyNodeRepositoryAction {
-
-    private static final Logger logger = LogManager.getLogger(VerifyNodeRepositoryAction.class);
+public class VerifyNodeRepositoryAction extends ActionType<ActionResponse.Empty> {
 
     public static final String ACTION_NAME = "internal:admin/repository/verify";
+    public static final VerifyNodeRepositoryAction INSTANCE = new VerifyNodeRepositoryAction();
```

Comment:
No need to subclass `ActionType` here (or anywhere really)

```suggestion
public class VerifyNodeRepositoryAction {

    public static final String ACTION_NAME = "internal:admin/repository/verify";
    public static final ActionType<ActionResponse.Empty> INSTANCE = new ActionType<>(ACTION_NAME);
```

although in fact since this is never called via the `Client`, only via the `TransportService`, it could be instantiated by `VerifyNodeRepositoryCoordinationAction` directly and doesn't need an `ActionType` at all.

Reply:
The action type is still needed for registration in ActionModule, but I've removed the subclassing.

Reply:
Right but why does it need to be registered in the `ActionModule` if it's never called via a `Client`? It's like the node-level sub-action in a `TransportNodesAction` which the coordinating action registers directly with the transport service:

https://github.com/elastic/elasticsearch/blob/6b419c13d318a14d33fdfdc9f209654db4035bcc/server/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java#L84